### PR TITLE
Code splitting plus webpack 4

### PR DESCRIPTION
**Pull request by haideralishah**
`React.lazy()` Implement the whole project components Mobile and Web both.

This also upgrades to webpack 4.

### CODE REVIEW

**Comment by norbertschuler** [comment]
As this solution loads the pages faster, but therefore for the first time changing between different pages has another short period of loading we should discuss which loading indicator to use: (1) current text "Loading...", our component `LoadingIndicator`, a newer less annoying loading indicator or maybe nothing at all showing just a blank screen?

**Comment by norbertschuler** [comment]
- [ ] Merge latest develop branch and also adapt the new code not yet converted.

**Comment by haideralishah** [comment]
Ok, I merge the latest code from `develop` to `code_spliting` branch and add LoadingIndicator Components.

**Comment by norbertschuler** [comment]
> Ok, I merge the latest code from `develop` to `code_spliting` branch and add LoadingIndicator Components.

Thanks. Merging will need some time to be sure to be correct (also please use `npm run lint:errors`). I am not sure which is the best kind of loading indicator, but I am not a designer ;-)

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "@bugsnag/js": "^6.4.3",
     "@haskkor/react-native-recaptchav3": "1.0.3",
     "@react-native-community/async-storage": "^1.6.2",
+    "array-unique": "^0.3.2",
```
Comment (norbertschuler) [inline package.json]: I removed this unused npm module before. Must have come back by the merge. Do you really need this - for what purpose?

File: package.json [hunk 2]
```diff
@@ -39,12 +40,16 @@
     "esri-loader": "^2.12.0",
     "formik": "^1.5.8",
     "history": "^4.10.1",
+    "g": "^2.0.1",
```
Comment (norbertschuler) [inline package.json]: Unused?

File: package.json [hunk 3]
```diff
@@ -39,12 +40,16 @@
...
     "i18next": "^11.3.1",
     "intl": "1.2.5",
     "jwt-decode": "^2.2.0",
     "liform-react": "^0.9.0",
     "lodash": "^4.17.10",
+    "mini-css-extract-plugin": "^0.8.0",
+    "node": "^10.16.3",
```
Comment (norbertschuler) [inline package.json]: I think we don't need to install node with npm?

File: package.json [hunk 4]
```diff
@@ -39,12 +40,16 @@
...
     "normalizr": "^3.2.4",
+    "npm": "^6.13.0",
```
Comment (norbertschuler) [inline package.json]: I think we don't need to install npm with npm?

File: package.json [hunk 5]
```diff
@@ -53,6 +58,7 @@
     "react-async-script-loader": "^0.3.0",
     "react-autosuggest": "^9.3.4",
     "react-datepicker": "^2.9.6",
+    "react-dev-utils": "^9.1.0",
```
Comment (norbertschuler) [inline package.json]: Is this used? And if not why not added in devDependencies?

File: package.json [hunk 6]
```diff
@@ -115,8 +121,10 @@
     "tcomb-json-schema": "^0.3.2",
     "tipsi-stripe": "^5.6.0",
     "uuid": "^3.3.3",
+    "ts-loader": "^6.2.1",
```
Comment (norbertschuler) [inline package.json]: Unused?

File: package.json [hunk 7]
```diff
@@ -115,8 +121,10 @@
...
     "uuidv4": "^2.0.0",
     "why-did-you-update": "1.0.6",
+    "yargs": "^14.2.0",
```
Comment (norbertschuler) [inline package.json]: Unused?

File: package.json [hunk 8]
```diff
@@ -136,16 +144,18 @@
     "eslint-plugin-json": "1.4.0",
     "eslint-plugin-react": "^7.16.0",
     "eslint-plugin-react-native": "^3.8.1",
-    "extract-text-webpack-plugin": "^3.0.2",
     "favicons-webpack-plugin": "^1.0.2",
-    "file-loader": "^1.1.8",
+    "file-loader": "^4.2.0",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
+    "fork-ts-checker-notifier-webpack-plugin": "^1.0.2",
```
Comment (norbertschuler) [inline package.json]: Unused?

File: package.json [hunk 9]
```diff
@@ -154,11 +164,15 @@
     "reactotron-react-native": "^3.7.1",
     "sass-loader": "^6.0.6",
     "style-loader": "^0.20.1",
-    "url-loader": "^0.6.2",
-    "webpack": "^3.12.0",
+    "uglifyjs-webpack-plugin": "^2.2.0",
```
Comment (norbertschuler) [inline package.json]: Unused?

File: package.json (hunk 1 above)
Comment (haideralishah) [reply package.json]: I didn't add this module `array-unique`.

Comment (norbertschuler) [inline package.json]: I removed this unused npm module before. Must have come back by the merge. Do you really need this - for what purpose?

**Comment by norbertschuler** [comment]
@haideralishah should we close this in favor or #1647 and #1650?

**Pull request closed without merging**